dissect/target/helpers/configparser.py

```diff
@@ -0,0 +1,241 @@
+
+        for key, value in dictionary.items():
+            value = self._collapse_dict(value, self.collapse_all or key in self.collapse)
+            new_dictionary.update({key: value})
+
+        return new_dictionary
+
+
+class Ini(ConfigurationParser):
+    def __init__(
+        self,
+        collapse: Optional[Union[bool, set]] = True,
+        seperator: tuple[str] = ("=",),
+        comment_prefixes: tuple[str] = (";", "#"),
+    ) -> None:
+        super().__init__(
+            collapse,
+            seperator=seperator,
+            comment_prefixes=comment_prefixes,
+        )
+
+        self.parsed_data = ConfigParser(
+            strict=False,
+            delimiters=self.seperator,
+            comment_prefixes=self.comment_prefixes,
+        )
+        self.parsed_data.optionxform = str
+
+    def parse_file(self, fh: io.TextIO) -> None:
+        offset = fh.tell()
+        try:
+            self.parsed_data.read_file(fh)
+            return
+        except MissingSectionHeaderError:
+            pass
+
+        fh.seek(offset)
+        open_file = io.StringIO("[DEFAULT]\n" + fh.read())
+        self.parsed_data.read_file(open_file)
+
+
+class Txt(ConfigurationParser):
+    """Read the file into ``content``, and show the bumber of bytes read."""
+
+    def parse_file(self, fh: TextIO) -> None:
+        # Cast the size to a string, to print it out later.
+        self.parsed_data = {"content": fh.read(), "size": str(fh.tell())}
+
+
+class Default(ConfigurationParser):
+    """Parse a configuration file specified by ``seperator`` and ``comment_prefixes``.
+
+    This parser does the following things:
+
+        key<seperator>value    -> {"key": "value"}
+        <empty_space><comment> -> skip
+    """
+
+    def __init__(
+        self,
+        collapse: Optional[Union[bool, set]] = False,
+        seperator: tuple[str] = (r"=",),
+        comment_prefixes: tuple[str] = (";", "#"),
+    ) -> None:
+        super().__init__(collapse, seperator, comment_prefixes)
+        self.SEPERATOR = re.compile(rf"\s*[{''.join(seperator)}]\s*")
+        self.COMMENTS = re.compile(rf"\s*[{''.join(comment_prefixes)}]")
+
+    def parse_file(self, fh: TextIO) -> None:
+        information_dict = {}
+
+        skip_lines = self.comment_prefixes + ("\n",)
+        prev_key = (None, None)
+        for line in fh.readlines():
+            if line.strip().startswith(skip_lines):
+                continue
+
+            # Strip the comments first
+            line, *_ = self.COMMENTS.split(line)
+
+            if not line.strip():
+                # The line was empty
+                continue
+
+            if line.startswith((" ", "\t")) and line.strip():
+                # This part was indented so it is a continuation of the previous key
+                prev_value = information_dict.get(prev_key)
+                if not isinstance(prev_value, list):
+                    prev_value = [prev_value]
+                prev_value.append(line.strip())
```

Comment:
The idea is to parse single key, single value? So a continuation value should be simply string appended to the existing value?